openwisp_controller/config/api/filters.py

```diff
@@ -0,0 +1,119 @@
+from uuid import UUID
+
+from django.utils.translation import gettext_lazy as _
+from django_filters import rest_framework as filters
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework.exceptions import ValidationError
+from swapper import load_model
+
+from openwisp_controller.geo.api.views import BaseOrganizationFilter
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+DeviceGroup = load_model('config', 'DeviceGroup')
+
+
+class BaseConfigAPIFilter(BaseOrganizationFilter):
+    def _set_valid_filterform_lables(self):
+        # When not filtering on a model field, an error message
+        # with the label "[invalid_name]" will be displayed in filter form.
+        # To avoid this error, we need to provide the label explicitly.
+        raise NotImplementedError
+
+
+class TemplateListFilter(BaseConfigAPIFilter):
+    class Meta:
+        model = Template
+        fields = {
+            'backend': ['exact'],
+            'type': ['exact'],
+            'default': ['exact'],
+            'required': ['exact'],
+            'created': ['exact', 'gte', 'lt'],
+        }
+
+
+class VPNListFilter(BaseConfigAPIFilter):
+
+    subnet = filters.CharFilter(field_name='subnet', label=_('VPN Subnet'))
+
+    class Meta:
+        model = Vpn
+        fields = {
+            'backend': ['exact'],
+            'subnet': ['exact'],
+        }
+
+
+class DeviceListFilterBackend(DjangoFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        """
+        Validate that the request parameters contain
+        a valid configuration template uuid format
+        """
+        config_template_uuid = request.query_params.get('config__templates')
+        if config_template_uuid:
+            try:
+                # Attempt to convert the uuid string to a UUID object
+                config_template_uuid_obj = UUID(config_template_uuid)
+            except ValueError:
+                raise ValidationError({'config__templates': 'Invalid UUID format'})
```

Comment:
remember to flag user facing strings as translatable, **always**.

Reply:
noted.